COMMITTERS.adoc

```diff
@@ -0,0 +1,173 @@
+
+== Merging Pull Requests
+
+The first thing is that, while being a committer,
+you can't really commit directly to the main repository:
+you need at least one review by your peers.
+
+Thus the need for a pull request.
+
+So you are more like a merger than a committer
+and you might merge more PRs from others than your own work.
+
+While not absolute, here are some advices:
+
+* if you really want a review by a specific person,
+  make it a named reviewer via a GitHub UI
+* if someone asked for a specific reviewer, better
+  give some time for this person to review the PR.
+  If not reviewed in a reasonable amount of time,
+  better ping them again to have some information
+  or ask the original author if we should go ahead.
+* if in doubt (on a specific part or because you're
+  not comfortable reviewing documentation or any other reason),
+  ask for help
+* except if it's a totally non brainer (typically a typo fix),
+  let the PR bake a few hours so some other people can have a
+  look if interested.
+  If the PR is very large or is a new extension, better let it
+  bake for a few days.
+* we do merge commits so please ask the author to properly
+  squash the commits before merging.
+  The idea is to have proper semantic commits.
+  If the author is not familiar with Git or not available,
+  you can also (carefully) do it for them as soon as you agreed on it.
+* make sure the commit comments are meaningful
+* at the beginning, you can start by giving your approval and not merging
+  and see how things evolve.
+  This should give you some hints of important things to check in the future.
+* only merge pull requests that target master.
+  **Dealing with old branches is a separate process.**
+* if you are sure a PR should go in and is just waiting for CI,
+  just use the `triage/waiting-for-ci` label.
+  It is a good hint for other committers.
+* if you think a pull request is an essential part of a given milestone
+  (and can realistically be part of it),
+  you should affect this milestone to it so that we can be sure it stays on the radar.
+
+Obviously, each situation is different so use your own judgement,
+and if in doubt, just ask for advice, the other committers are here to help.
+
+== Pull Request Title
+
+The title should be in English, should not contain an issue number.
+It should also not contain ellipsis.
+
+If your commit message was too long and GitHub automatically cut
+the title, that's helpful if you can take the time to move the cut
+part where it belongs to have a full title.
+
+Titles are included in the Release notes so they are important.
+
+A good title would look like: `Fix XXX in Quartz extension` or
+`Introduce Hibernate Reactive extension`.
+
+A bad title would look like: `fix(#444)`.
+
+== Issues Fixed
+
+When a PR fixes some issues, it's good practice to add it in the description.
+
+One issue per line with something like:
+
+[source,asciidoc]
+----
+* Fix #444
+* Fix #555
```

Comment:
```suggestion
* Resolves #555
```

Reply:
Yeah, there are a couple of different options but I'm not sure it's worth making the message more complex here.

Reply:
I added a TIP below.

Reply:
should it have description of what it fixes? not just the number?

Reply:
it would hopefully just be redundant. the "Fix #nnn" is mainly just to autoclose issues thus doesnt require more info by default

Reply:
If you pass your mouse on the issue, you have the issue info so it's perfectly OK as is.